cmds/tui.c

```diff
@@ -1106,10 +1116,60 @@ static void print_graph_empty(void)
 	if (list_empty(&graph_output_fields))
 		return;
 
-	list_for_each_entry(field, &graph_output_fields, list)
+	list_for_each_entry(field, &graph_output_fields, list) {
 		printw("%*s", field->length + FIELD_SPACE, "");
+		field_width += field->length + FIELD_SPACE;
+	}
 
 	printw(FIELD_SEP);
+	field_width += FIELD_SPACE;
+}
+
+static bool is_boxchar(unsigned char p)
+{
+	/*
+	 * simply check whether the given character is a box character or not.
+	 *   ├ = 0xe2949c, │ = 0xe29482, └ = 0xe29494,
+	 *   ─ = 0xe29480, ▶ = 0xe296b6
+	 */
+	if (p >= (unsigned char)0x80 && p <= (unsigned char)0xb6)
+		return true;
+	else
+		return false;
+}
+
+static void println(const char *fmt, ...)
+{
+	va_list ap;
+	va_start(ap, fmt);
+	char *apstr = NULL;
+	char *linestr = NULL;
+
+	xvasprintf(&apstr, fmt, ap);
+	xasprintf(&linestr, "%s%s", linebuf, apstr);
+	strncpy(linebuf, linestr, sizeof(linebuf));
```

Comment:
So is this for adding new formatted strings to the linebuf?

Reply:
Right.